apps/cowswap-frontend/src/modules/tradeFormValidation/services/validateTradeForm.ts

```diff
@@ -70,12 +71,14 @@ export function validateTradeForm(context: TradeFormValidationContext): TradeFor
     }
   }
 
-  if (!inputCurrencyBalance) {
-    return TradeFormValidation.BalancesNotLoaded
-  }
+  if (!isZeroBalanceOrderAllowed) {
+    if (!inputCurrencyBalance) {
+      return TradeFormValidation.BalancesNotLoaded
+    }
 
-  if (inputCurrencyBalance.lessThan(inputCurrencyAmount)) {
-    return TradeFormValidation.BalanceInsufficient
+    if (inputCurrencyBalance.lessThan(inputCurrencyAmount)) {
+      return TradeFormValidation.BalanceInsufficient
+    }
```

Comment:
quite a small change! 
I was expecting more noise :)

Reply:
All thanks to @shoom3301's magic ✨